pkg/roll/execute.go

```diff
@@ -15,7 +15,23 @@ import (
 	"github.com/xataio/pgroll/pkg/schema"
 )
 
+func (m *Roll) Validate(ctx context.Context, migration *migrations.Migration) error {
+	if m.skipValidation {
+		return nil
+	}
+	lastSchema, err := m.state.LastSchema(ctx, m.schema)
+	if err != nil {
+		return err
+	}
+	err = migration.Validate(ctx, lastSchema)
+	if err != nil {
+		return fmt.Errorf("migration '%s' is invalid: %w", migration.Name, err)
+	}
+	return nil
+}
+
 // Start will apply the required changes to enable supporting the new schema version
+// Migrations must be validated before running Start.
```

Comment:
How about having this `Start` method also perform validation, even though we have broken validation out into a separate step?

Previously the API of this package was

```
Start
Complete / Rollback
```

Now consumers need to remember to validate first:

```
Validate
Start
Complete / Rollback
```

Not sure there is a use-case where you'd deliberately want to Start without validating first.

WDYT?

Reply:
Works for me.